invenio_jsonschemas/views.py

```diff
@@ -45,10 +48,26 @@ def create_blueprint(state):
     @blueprint.route("/<path:schema_path>")
     def get_schema(schema_path):
         """Retrieve a schema."""
-        try:
-            return send_from_directory(state.get_schema_dir(schema_path),
-                                       schema_path)
-        except JSONSchemaNotFound:
-            abort(404)
+        replace_refs = request.args.get('refs', '')
+        if replace_refs == 'resolve':
+            schema = {'$ref': request.base_url}
+            loader_cls = import_string(
+                current_app.config['JSONSCHEMAS_LOADER_CLS']
+            )
+            try:
+                return jsonify(JsonRef.replace_refs(
+                    schema,
+                    loader=loader_cls())
+                )
+            except JsonRefError:
+                # jsonref will raise JsonRefError if the requested schema
+                # does not exist.
+                abort(404)
```

Comment:
`send_from_directory(state.get_schema_dir(schema_path), schema_path)` gives 404 as well. This is for the case when `<path:schema_path>` does not exist.